core/frontend/services/theme-engine/middleware/update-global-template-options.js

```diff
@@ -28,7 +28,9 @@ function calculateLegacyPriceData(products) {
         };
     }
 
-    const defaultProduct = products[0] || {};
```

Comment:
I think we use the first product as the default product elsewhere too, in the Members repo - would you mind having a check for that and updating it anywhere?

Reply:
Yeah definitely, I am actively testing and updating all places the default product could be a problem with this change, will update anywhere I find in Members repo as well 👍